repository_service_tuf_worker/repository.py

```diff
@@ -628,6 +633,34 @@ def bootstrap(
 
         return asdict(result)
 
+    def update_settings(
+        self,
+        payload: Dict[str, Any],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ):
+        """Update repository settings with the new settings."""
+        tuf_settings: Dict[str, Any] = payload.get("settings")
+        if tuf_settings is None:
+            raise KeyError("No 'settings' in the payload")
+
+        if tuf_settings.get("expiration") is None:
+            raise KeyError("No 'expiration' in the payload")
+
+        if len(tuf_settings["expiration"]) < 1:
+            raise KeyError("No role provided for expiration policy change")
+
+        logging.info("Updating settings")
+        online_roles = Roles.online_roles()
+        for role in tuf_settings["expiration"]:
```

Comment:
Following the implementation, it will case that some expiration can be updated and others roles not if it receives a payload with an invalid role.
It brings a problem to the user because the task will fail in the middle with partial change.

IMO we could apply the same behavior we do deleting targets (artifacts)
When we return the task, we say which ones are changed and which are invalid.